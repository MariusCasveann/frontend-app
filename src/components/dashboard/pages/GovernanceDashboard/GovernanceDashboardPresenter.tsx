import React, { useState } from 'react';
import { CardLayout } from '../../../common/molecules/CardLayout/CardLayout';
import GoogleCostPresenter from '../../molecules/GoogleCost/GoogleCostPresenter';
import DeploymentFrequencyPresenter from '../../molecules/DeploymentFrequency/DeploymentFrequencyPresenter';
import GovernanceFilterPresenter from './GovernanceFilterPresenter';
import { fiscalYear, fiscalYearEnd, fiscalYearStart } from '../../../../utils/fiscalYear';
import DomainCostPresenter from '../../molecules/DomainCost/DomainCostPresenter';
import '../common/Dashboard.css';
import { FiltersContext } from '../../atoms/Filters/FiltersContext';
import { MESSAGE_NO_DATA_AVAILABLE_SELECT_ANOTHER_FINANCIAL_YEAR } from '../../../../utils/constants';

export default () => {
    const [fromDateState, setFromDate] = useState<string>(fiscalYearStart);
    const [toDateState, setToDate] = useState<string>(fiscalYearEnd);
    const [financialYearState, setFinancialYear] = useState<string | null>(fiscalYear.toString());

    return (
        <FiltersContext.Provider
            value={{
                fromDate: fromDateState,
                toDate: toDateState,
                setToDate,
                setFromDate,
                financialYear: financialYearState,
                setFinancialYear,
                messageWhenNoData: MESSAGE_NO_DATA_AVAILABLE_SELECT_ANOTHER_FINANCIAL_YEAR
            }}
        >
            <div className="dashboard-wrapper">
                <GovernanceFilterPresenter />
                <CardLayout width={100} className="dashboard-card-layout">
                    <GoogleCostPresenter />
                    <DomainCostPresenter />
                    <DeploymentFrequencyPresenter />
                </CardLayout>
            </div>
        </FiltersContext.Provider>
    );
};
