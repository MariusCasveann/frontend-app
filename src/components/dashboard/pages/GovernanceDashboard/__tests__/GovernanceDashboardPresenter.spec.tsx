import * as React from 'react';
import renderer from 'react-test-renderer';
jest.mock('../../../../common/molecules/CardLayout/CardLayout', () => ({ Card: 'Card', CardLayout: 'CardLayout' }));
jest.mock('../../../molecules/GoogleCost/GoogleCostPresenter', () => 'GoogleCostPresenter');
jest.mock('../../../molecules/DeploymentFrequency/DeploymentFrequencyPresenter', () => 'DeploymentFrequencyPresenter');
jest.mock('../../../molecules/DomainCost/DomainCostPresenter', () => 'DomainCostPresenter');
jest.mock('../GovernanceFilterPresenter', () => 'GovernanceFilterPresenter');
jest.doMock('../../../atoms/Filters/FiltersContext', () => ({
    FiltersContext: 'FiltersContext',
    'FiltersContext.Consumer': 'FiltersContext.Consumer'
}));

import GovernanceDashboardPresenter from '../GovernanceDashboardPresenter';

describe(' GovernanceDashboardPresenter', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<GovernanceDashboardPresenter />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
