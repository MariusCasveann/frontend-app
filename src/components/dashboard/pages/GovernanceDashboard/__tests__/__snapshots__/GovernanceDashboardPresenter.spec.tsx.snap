// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` GovernanceDashboardPresenter renders correctly 1`] = `
<div
  className="dashboard-wrapper"
>
  <GovernanceFilterPresenter />
  <CardLayout
    className="dashboard-card-layout"
    width={100}
  >
    <GoogleCostPresenter />
    <DomainCostPresenter />
    <DeploymentFrequencyPresenter />
  </CardLayout>
</div>
`;
